File: src/python/core/src/dicebear/style_def/meta_license.py
"""Read-only view over the ``meta.license`` block of a style definition."""

from __future__ import annotations

from typing import Any, cast


class MetaLicense:
    """Read-only view over the ``meta.license`` block of a style definition."""

    def __init__(self, data: dict[str, Any]) -> None:
        self._data = data

    def name(self) -> str | None:
        """Return the license name (e.g. ``"CC BY 4.0"``), or ``None``."""
        return cast("str | None", self._data.get("name"))

    def url(self) -> str | None:
        """Return the license URL, or ``None`` when not set."""
        return cast("str | None", self._data.get("url"))

    def text(self) -> str | None:
        """Return the full license text, or ``None`` when not set."""
        return cast("str | None", self._data.get("text"))

File: src/csharp/core/src/Style/MetaLicense.cs
using System.Text.Json.Nodes;

namespace DiceBear.Internal
{
    /// <summary>
    /// Read-only view over the <c>meta.license</c> block of a style
    /// definition.
    /// </summary>
    internal sealed class MetaLicense
    {
        private readonly JsonObject _data;

        internal MetaLicense(JsonObject data) => _data = data;

        /// <summary>
        /// Returns the license name (e.g. <c>"CC BY 4.0"</c>), or
        /// <see langword="null"/> when not set.
        /// </summary>
        internal string? Name() => JsonRead.Str(_data, "name");

        /// <summary>
        /// Returns the license URL, or <see langword="null"/> when not set.
        /// </summary>
        internal string? Url() => JsonRead.Str(_data, "url");

        /// <summary>
        /// Returns the full license text, or <see langword="null"/> when not
        /// set.
        /// </summary>
        internal string? Text() => JsonRead.Str(_data, "text");
    }
}
